app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt

```diff
@@ -115,32 +93,95 @@ fun SurveyDetailScreen(
                 scale = ZOOMED_IN_SCALE
             }
         }.start()
+        viewModel.getSurveyQuestions(surveyId = survey.id)
     }
 
-    val popBackAnimationTimer = object : CountDownTimer(ZOOM_DURATION_IN_MILLIS, ZOOM_INTERVAL_IN_MILLIS) {
-        override fun onTick(millisUntilFinished: Long) {
-            scale -= SCALE_PER_COUNTDOWN
-        }
+    fun zoomOut() {
+        object : CountDownTimer(ZOOM_DURATION_IN_MILLIS, ZOOM_INTERVAL_IN_MILLIS) {
+            override fun onTick(millisUntilFinished: Long) {
+                scale -= SCALE_PER_COUNTDOWN
+            }
 
-        override fun onFinish() {
-            scale = ORIGINAL_SCALE
-            navigator.popBackStack()
-        }
+            override fun onFinish() {
+                scale = ORIGINAL_SCALE
+                navigator.popBackStack()
+            }
+        }.start()
     }
 
     BackHandler {
         if (currentPage > 0) showConfirmDialog = true
-        else popBackAnimationTimer.start()
+        else zoomOut()
     }
 
-    LaunchedEffect(key1 = pagerState) {
-        snapshotFlow { pagerState.currentPage }.collect { page ->
-            viewModel.setCurrentPage(page)
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+        SurveyQuestionDetailContent(
+            survey = survey,
+            surveyQuestions = surveyQuestions,
+            currentPage = currentPage,
+            scale = scale,
+            onSetCurrentPage = {
+                viewModel.setCurrentPage(it)
+            },
+            onSubmitSurvey = {
+                viewModel.submitSurvey(surveyId = survey.id)
+            },
```

Comment:
These are quite short, we could put them on one line 😄 
```suggestion
            onSetCurrentPage = { viewModel.setCurrentPage(it) },
            onSubmitSurvey = { viewModel.submitSurvey(surveyId = survey.id) },
```